crates/core/src/error/summary.rs

```diff
@@ -0,0 +1,592 @@
+            // We ignore the result here, as we don't want to propagate the error
+            _ = Self::log_issue(scope, category, &message, &root_cause);
+        }
+
+        _ = self
+            .issues
+            .entry(category)
+            .or_default()
+            .entry(scope)
+            .and_modify(|val| {
+                val.count += 1;
+                if val.root_cause.is_none() {
+                    val.root_cause.clone_from(&root_cause);
+                }
+            })
+            .or_insert(CondensedIssue {
+                category,
+                message,
+                count: 1,
+                root_cause,
+            });
+    }
+
+    /// Adds a custom metric
+    pub fn add_metric(&mut self, key: impl Into<EcoString>, value: impl Into<EcoString>) {
+        let value = value.into();
+
+        _ = self
+            .metrics
+            .entry(key.into())
+            .and_modify(|val| *val = value.clone())
+            .or_insert_with(|| value);
+    }
+
+    pub fn export_issues(&mut self) -> bool {
+        self.display.insert(DisplayOptionKind::Issues)
+    }
+
+    pub fn export_timing(&mut self) -> bool {
+        self.display.insert(DisplayOptionKind::Timing)
+    }
+
+    pub fn export_metrics(&mut self) -> bool {
+        self.display.insert(DisplayOptionKind::Metrics)
+    }
+
+    pub fn export_all(&mut self) -> bool {
+        self.display.insert(DisplayOptionKind::All)
+    }
+
+    pub fn export_none(&mut self) {
+        self.display.clear();
+    }
+
+    pub fn set_export(&mut self, option: DisplayOptionKind) -> bool {
+        self.display.clear();
+        self.display.insert(option)
+    }
+
+    fn log_issue(
+        scope: IssueScope,
+        category: IssueCategory,
+        message: &EcoString,
+        root_cause: &Option<EcoString>,
+    ) -> fmt::Result {
+        let mut to_print = String::new();
+
+        if root_cause.is_none() {
+            writeln!(to_print, "in scope '{scope}': {message}",)?;
+        } else {
+            let root_cause = format_root_cause(root_cause);
+            writeln!(to_print, "in scope '{scope}': {message}{root_cause}",)?;
+        }
+
+        match category {
+            IssueCategory::Error => log::error!("{to_print}"),
+            IssueCategory::Warning => log::warn!("{to_print}"),
+            IssueCategory::Info => log::info!("{to_print}"),
+        }
+
+        Ok(())
+    }
+
+    // ! TODO: Refactor this method to merge the context of each summary
+    // !
+    // ? How do we merge the context of each summary?
+    // ? We can't merge the context, as it's a unique identifier for each summary.
+    // ? We could add a new field to each CondensedIssue to store the context of the
+    // ? merged summaries.
+    // ?
+    // ? How do we merge the other fields? E.g. Timing, Metrics, etc.
+    pub fn merge(&mut self, other: Self) {
+        self.issues.extend(other.issues);
+        self.metrics.extend(other.metrics);
+    }
```

Comment:
Use `conflate` and implement `Merge`, make sure we fulfil above requirements and make it as ergonomic as possible. Make sure deduplication works well also for edge cases. We want to be sure, that we can assess if a command soft errored (warnings are lower priority).

Reply:
Ah, conflate is actually behind the `merge` feature flag and doesn't make sense then to be used here.